hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/handle/HandleWorkflowTest.java

```diff
@@ -495,6 +529,30 @@ void testPreHandleWithDueDiligenceFailure() {
         // TODO: Verify that we created a penalty payment (https://github.com/hashgraph/hedera-services/issues/6811)
     }
 
+    @Test
+    @DisplayName("Test when there is the same transaction in the cache")
+    void testExistingTransactionInCache() {
+        // given
+        when(hederaRecordCache.getRecord(any())).thenReturn(generateDummyTransactionRecord());
+
+        // when
+        workflow.handleRound(state, round);
+
+        // then
+        assertThat(accountsState.isModified()).isFalse();
+        assertThat(aliasesState.isModified()).isFalse();
+        verify(recordManager, times(1)).startUserTransaction(any());
+        verify(recordManager, times(1)).endUserTransaction(any());
+    }
+
+    private TransactionRecord generateDummyTransactionRecord() {
+        final var body = new OneOf<>(BodyOneOfType.UNSET, 1);
+        final var entropy = new OneOf<>(EntropyOneOfType.PRNG_BYTES, 1);
+        return new TransactionRecord(
+                null, null, null, null, null, 1L, body, null, null, null, null, null, null, null, null, null, entropy,
+                null);
+    }
```

Comment:
It would be good to add a test that doesn't use a mocked HederaRecordCache (or at least, one that remembers values it was previously called with), and actually run two duplicate transactions and make sure the first one goes through fine, and the second one throws a duplicate exception. This particular test, for example, didn't catch the fact that the code was missing the `add` call, whereas the test I'm suggesting would have caught it.